src/watts/plugin_dakota.py

```diff
@@ -0,0 +1,268 @@
+    template_file
+        Templated Dakota input
+    extra_inputs
+        List of extra (non-templated) input files that are needed
+    extra_template_inputs
+        Extra templated input files
+    show_stdout
+        Whether to display output from stdout when Dakota is run
+    show_stderr
+        Whether to display output from stderr when Dakota is run
+
+    """
+
+    def __init__(self, template_file: str,  
+                 extra_inputs: Optional[List[str]] = None,
+                 extra_template_inputs: Optional[List[PathLike]] = None,
+                 auto_link_files: Optional[str] = None,
+                 show_stdout: bool = False, show_stderr: bool = False):
+        super().__init__(template_file, extra_inputs, extra_template_inputs,
+                         show_stdout, show_stderr)
+
+        dakota_dir = Path(os.environ.get("DAKOTA_DIR", ""))
+        self._executable = dakota_dir / f"dakota.sh"
+        self.input_name = template_file
+        self._auto_link_files = auto_link_files
+
+        # Setup to automatically include all 'extra_inputs' and 'extra_template_inputs' 
+        # to Dakota's "link files" option. Create a string of the file names in
+        # 'extra_inputs' and 'extra_template_inputs'.
+        if self._auto_link_files is not None:
+            dakota_link_files = []
+            if extra_inputs is not None:
+                for item in extra_inputs:
+                    dakota_link_files.append(item)
+
+            if extra_template_inputs is not None:
+                for item in extra_template_inputs:
+                    dakota_link_files.append(item)
+
+            self.dakota_link_files_string = "'"
+            for item in dakota_link_files:
+                self.dakota_link_files_string = f"{self.dakota_link_files_string}{item}' '"
+
+    def prerun(self, params: Parameters, filename: Optional[str] = None):
+        """ Change the permisison of the Dakota driver file
+
+        Parameters
+        ----------
+        params
+            Parameters used to render template
+        """
+
+        # Store the string for the "link_files" in params.
+        if self._auto_link_files is not None:
+            params[self._auto_link_files] = self.dakota_link_files_string
+
+        super().prerun(params)
+        if 'dakota_driver_name' in params.keys():
+            os.chmod(params['dakota_driver_name'], 0o755)
+
+    @property
+    def execute_command(self):
+        return [str(self.executable), "-i", self.input_name]
+
+
+def run_dakota_driver(coupled_code_exec: str):
+    """ Function to execute the workflow for data
+    exchange between Dakota and the coupled code
+    in the Dakota driver script.
+
+    Parameters
+    ----------
+    coupled_code_exec
+        The name of the WATTS python script of the
+        coupled code.
+    """
+    results = parse_dakota_input()
+    retval = run_coupled_code(coupled_code_exec)
+    return_dakota_input(results, retval)
+
+def parse_dakota_input():
```

Comment:
If this (and the other functions called from `run_dakota_driver`) are not meant to be user-facing, they should be preceded by a single underscore (i.e., change the name to `_parse_dakota_input`)